DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/SyntaxNodeUtils.cs

```diff
@@ -0,0 +1,178 @@
+                if (attributeLists.Any())
+                {
+                    foreach (var attributeList in attributeLists)
+                    {
+                        var attribute = attributeList.ChildNodes().First();
+                        if (attribute.ChildNodes().First().ToString().Equals("FunctionName"))
+                        {
+                            functionAttribute = attribute;
+                            return true;
+                        }
+                    }
+                }
+            }
+
+            functionAttribute = null;
+            return false;
+        }
+        
+        public static bool IsMarkedDeterministic(SyntaxNode node)
+        {
+            if (GetDeterministicAttribute(node) != null)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public static SyntaxNode GetDeterministicAttribute(SyntaxNode node)
+        {
+            if (TryGetMethodDeclaration(out SyntaxNode methodDeclaration, node))
+            {
+                var IEnumeratorAttributeList = methodDeclaration.ChildNodes().Where(x => x.IsKind(SyntaxKind.AttributeList));
+                if (IEnumeratorAttributeList.Any())
+                {
+                    foreach (SyntaxNode attributeList in IEnumeratorAttributeList)
+                    {
+                        if (attributeList.ToString().Equals("[Deterministic]"))
+                        {
+
+                            return attributeList;
+                        }
+                    }
+                }
+                
+            }
+            
+            return null;
+        }
+
+        internal static bool TryGetParameterNodeNextToAttribute(out SyntaxNode inputType, AttributeSyntax attributeExpression, SyntaxNodeAnalysisContext context)
+        {
+            var parameter = attributeExpression.Parent.Parent;
+            var parameterTypeNamesEnumerable = parameter.ChildNodes().Where(x => x.IsKind(SyntaxKind.IdentifierName) || x.IsKind(SyntaxKind.PredefinedType));
+            if (parameterTypeNamesEnumerable.Any())
+            {
+                inputType = parameterTypeNamesEnumerable.First();
+                return true;
+            }
+
+            inputType = null;
+            return false;
+        }
+
+        internal static bool TryGetReturnType(out ITypeSymbol returnType, AttributeSyntax attributeExpression, SyntaxNodeAnalysisContext context)
+        {
+            if (TryGetMethodDeclaration(out SyntaxNode methodDeclaration, attributeExpression))
+            {
+                returnType = context.SemanticModel.GetTypeInfo((methodDeclaration as MethodDeclarationSyntax).ReturnType).Type;
+                return true;
+            }
+
+            returnType = null;
+            return false;
+        }
+
+        internal static bool TryGetMethodDeclaration(out SyntaxNode methodDeclaration, SyntaxNode node)
```

Comment:
NIT: In general, the out parameter of `TryGet` methods go last, and the collection/object you are retrieving the item from goes first.